gravitee-am-identityprovider/gravitee-am-identityprovider-mongo/src/main/java/io/gravitee/am/identityprovider/mongo/user/MongoUserProvider.java

```diff
@@ -258,4 +266,30 @@ private byte[] createSalt() {
         random.nextBytes(salt);
         return salt;
     }
+
+    /**
+     * Create or update index on username field
+     */
+    private void createOrUpdateIndex() {
+        if (ensureIndexOnStart) {
+            getUserNameIndex()
+                    .andThen(Completable.fromPublisher(usersCollection.createIndex(new Document(configuration.getUsernameField(), 1),
+                            new IndexOptions().name(INDEX_USERNAME_UNIQUE).unique(true))))
+                    .subscribe(
+                            () -> LOGGER.debug("Unique index on username created"),
+                            err -> LOGGER.error("An error occurred while creating index {} with unique constraints", INDEX_USERNAME, err));
```

Comment:
Maybe better to log with `INDEX_USERNAME_UNIQUE`

Reply:
done